File: paper/src/main/java/com/al3x/housing2/Data/ConditionalData.java
package com.al3x.housing2.Data;

import com.al3x.housing2.Condition.Condition;
import com.al3x.housing2.Condition.ConditionEnum;
import lombok.Getter;
import lombok.Setter;

import java.util.HashMap;
import java.util.List;
import java.util.ArrayList;

@Getter
@Setter
public class ConditionalData {
    private String condition;
    private HashMap<String, Object> data;

    public ConditionalData() {

    }

    public ConditionalData(String condition, HashMap<String, Object> data) {
        this.condition = condition;
        this.data = data;
    }

    public static List<ConditionalData> fromList(List<Condition> conditions) {
        List<ConditionalData> list = new ArrayList<>();
        for (Condition condition : conditions) {
            list.add(new ConditionalData(condition.getName(), condition.data()));
        }
        return list;
    }

    public static List<Condition> toList(List<ConditionalData> conditionList) {
        List<Condition> list = new ArrayList<>();
        for (ConditionalData data : conditionList) {
            ConditionEnum conditionEnum = ConditionEnum.getConditionByName(data.getCondition());
            if (conditionEnum == null) {
                throw new IllegalArgumentException("Condition " + data.getCondition() + " does not exist");
            }
            list.add(conditionEnum.getConditionInstance(data.getData()));
        }
        return list;
    }
}

File: paper/src/main/java/com/al3x/housing2/Instances/HousingData/ConditionalData.kt
package com.al3x.housing2.Instances.HousingData

import com.al3x.housing2.Condition.Condition
import com.al3x.housing2.Condition.ConditionEnum

data class ConditionData(
    val condition: String,
    val data: HashMap<String, Any>
) {
    companion object {
        fun fromList(conditions: List<Condition>): List<ConditionData> {
            val list = mutableListOf<ConditionData>()
            conditions.forEach {
                list.add(ConditionData(it.name, it.data()))
            }
            return list
        }

        fun toList(conditionList: List<ConditionData>): List<Condition> {
            val list = mutableListOf<Condition>()
            conditionList.forEach {
                if (ConditionEnum.getConditionByName(it.condition) == null) {
                    throw IllegalArgumentException("Condition ${it.condition} does not exist")
                }
                list.add(ConditionEnum.getConditionByName(it.condition)!!.getConditionInstance(it.data))
            }
            return list
        }
    }
}